server/src/test/scala/org/http4s/server/middleware/GZipSpec.scala

```diff
@@ -20,5 +25,21 @@ class GZipSpec extends Http4sSpec {
       resp.status must_== (Status.Ok)
       resp.headers.get(`Content-Encoding`) must beNone
     }
+
+    checkAll("encoding", new Properties("GZip") {
+      property("middleware encoding == GZIPOutputStream encoding") = forAll { value: String =>
+        val service = GZip(HttpService { case GET -> Root => Ok(value) })
+        val req = Request(Method.GET, Uri.uri("/")).putHeaders(`Accept-Encoding`(ContentCoding.gzip))
+        val actual = service.orNotFound(req).unsafeRun.body.runLog.unsafeRun.toArray
```

Comment:
`service.orNotFound(req).as[Array[Byte]]` will get you a `Task[Array[Byte]]`. Then on line 41, `actual must returnValue(expected)`.

No unsafeRun anywhere. :)